youtube_dl/extractor/funimation.py

```diff
@@ -157,8 +171,15 @@ def _real_extract(self, url):
                     webpage, 'players data'),
                 display_id)[0]['playlist']
 
-            items = next(item['items'] for item in playlist if item.get('items'))
-            item = next(item for item in items if item.get('itemAK') == display_id)
+            items = next(item for item in playlist if item.get('items'))
+            item = next(item for item in items['items'] if item.get('itemAK') == display_id)
+            if items.get('itemClass') == 'season':
+                season_id = items.get('itemAK')
+                season_number = int_or_none(self._search_regex(
+                    r'^Season ([0-9]+)$', season_id, 'season number', NO_DEFAULT, False))
+                episode_number = (int(float(item.get('number'))) -
+                                  int(float(items['items'][0].get('number'))) +
+                                  1)
```

Comment:
Do you think I should remove the calculation entirely? Because from what I can tell most Funimation series have the appropriate data, and it would be useful to have the season and episode number available. (I can just add a check and eliminate the number if the data isn't there.)